src/Tasks/NETMessage.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Tasks.Deployment.Bootstrapper;
+
+namespace Microsoft.Build.Tasks
+{
+    public sealed class NETMessage : TaskExtension
+    {
+        private readonly Dictionary<BuildMessageSeverity, Func<bool>> _severityToActionMap;
+
+        public NETMessage()
+        {
+            _severityToActionMap = new Dictionary<BuildMessageSeverity, Func<bool>>()
+            {
+                {
+                    BuildMessageSeverity.Error, () =>
+                    {
+                        Log.LogErrorWithCodeFromResources(ResourceName, FormatArguments);
+                        return false;
+                    }
+                },
+                {
+                    BuildMessageSeverity.Warning, () =>
+                    {
+                        Log.LogWarningWithCodeFromResources(ResourceName, FormatArguments);
+                        return true;
+                    }
+                },
+                {
+                    BuildMessageSeverity.Info, () =>
+                    {
+                        MessageImportance importance = (MessageImportance)Enum.Parse(typeof(MessageImportance), MessageImportance, true);
+                        Log.LogMessageFromResources(importance, ResourceName, FormatArguments);
+                        return true;
+                    }
+                },
+            };
+        }
```

Comment:
This seems unnecessarily complex. Why not inline it as a `switch` statement into `Execute`?